app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1265,22 +1265,18 @@ public void setSession(@NonNull Session aSession, @OldSessionDisplayAction int a
             mViewModel.setIsDesktopMode(mSession.getUaMode() == WSessionSettings.USER_AGENT_MODE_DESKTOP);
 
             if (hidePanel) {
-                if (oldSession != null) {
-                    onCurrentSessionChange(oldSession.getWSession(), aSession.getWSession());
-                } else {
-                    onCurrentSessionChange(null, aSession.getWSession());
-                }
+                hideNewTabPanel(true);
+                hideLibraryPanel(true);
+                onCurrentSessionChange((oldSession != null ? oldSession.getWSession() : null), aSession.getWSession());
             }
 
             for (WindowListener listener: mListeners) {
                 listener.onSessionChanged(oldSession, aSession);
             }
-        }
-        mCaptureOnPageStop = false;
-
-        if (hidePanel) {
+        } else if (hidePanel) {
```

Comment:
Why we don't need to close the library if the aSession is the same as the stored mSession ? Wouldn't this contradict the boolean passed as argument ? Wouldn't be better to just pass false in that argument for such scenario ?

Reply:
If the two sessions are the same, calling `closeLibrary()` is not likely to have side effects.

I am not so sure about the case when the two sessions are different, so we try to hide the library in a more careful way.